src/common/utxobased/info/scriptTemplates/types.ts

```diff
@@ -0,0 +1,14 @@
+export interface ScriptTemplates {
+  [key: string]: ScriptTemplate
+}
+
+export type ScriptTemplate = (
+  pubkey: string,
+  optionalArgs?: OptionalScriptTemplateArgs
+) => string
+
+export interface OptionalScriptTemplateArgs {
```

Comment:
No need for this type here, try and keep the files outside of the info folder as "generic" as possible.

Reply:
I could not come up with a better way to define these additional parameters, that does not involve either weakening the type, or a generic that would propagate the type selection up to the core.